esi_leap/api/controllers/v1/contract.py

```diff
@@ -137,10 +138,36 @@ def delete(self, contract_id):
 
         permitted.cancel()
 
+    @staticmethod
+    def _get_offer_by_uuid_or_name(uuid_or_name):
```

Comment:
I think `_get_offer_by_uuid_or_name` (and `_get_contract_by_uuid_or_name) might be called enough that you want to separate it out into a `utils.py` in `esi_leap/api/controllers/v1/utils.py`.

You will also definitely need tests for these functions.